src/box-node-sdk.ts

```diff
@@ -0,0 +1,463 @@
+/**
+ * @fileoverview Box SDK for Node.js
+ */
+
+import { EventEmitter } from 'events';
+import * as qs from 'querystring';
+import AnonymousAPISession = require('./sessions/anonymous-session');
+import APIRequestManager = require('./api-request-manager');
+import BoxClient = require('./box-client');
+import TokenManager = require('./token-manager');
+
+// ------------------------------------------------------------------------------
+// Typedefs and Callbacks
+// ------------------------------------------------------------------------------
+
+/**
+ * Object representing interface functions for PersistentClient to interact with the consumer app's central storage layer.
+ * @typedef {Object} TokenStore
+ * @property {ReadTokenInfoFromStore} read - read TokenInfo from app central store.
+ * @property {WriteTokenInfoToStore} write - write TokenInfo to the app's central store.
+ * @property {ClearTokenInfoFromStore} clear - delete TokenInfo from the app's central store.
+ */
+
+/**
+ * Acquires TokenInfo from the consumer app's central store.
+ * @typedef {Function} ReadTokenInfoFromStore
+ * @param {Function} callback - err if store read issue occurred, otherwise propagates a TokenInfo object
+ */
+
+/**
+ * Writes TokenInfo to the consumer app's central store
+ * @typedef {Function} WriteTokenInfoToStore
+ * @param {TokenInfo} tokenInfo - the token info to be written
+ * @param {Function} callback - err if store write issue occurred, otherwise propagates null err
+ *  and null result to indicate success
+ */
+
+/**
+ * Clears TokenInfo from the consumer app's central store
+ * @typedef {Function} ClearTokenInfoFromStore
+ * @param {Function} callback - err if store delete issue occurred, otherwise propagates null err
+ *  and null result to indicate success
+ */
+
+type TokenStore = object /* FIXME */;
+type UserConfigurationOptions = object /* FIXME */;
+type TokenRequestOptions = object /* FIXME */;
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+var Config = require('./util/config'),
+	BasicAPISession = require('./sessions/basic-session'),
+	PersistentAPISession = require('./sessions/persistent-session'),
+	AppAuthSession = require('./sessions/app-auth-session'),
+	Webhooks = require('./managers/webhooks');
```

Comment:
Should we move these requirements to be grouped with the imports on lines 5-10? We could move the "Requirements" header up there too